### PR TITLE
Part3

### PR DESCRIPTION
**Pull request by gib-bowden**

### CODE REVIEW

**Comment by rousell** [comment]
@gib-bowden seriously, good job on this. The one thing that I definitely want you to fix is that if I have only one contact saved for my user and I try to delete the contact, then it doesn't disappear from the page when I click the 'X' (though it is deleting in firebase). 

Additionally, it would be nice for there to be clearer notification for the user on which fields need to be updated for changes, and what the changes they need to do are. For the former, there are validation classes that are added to input fields that you can use to manipulate how the fields look visually. However, this is a 'nice to have' / isn't required to be implemented for this PR

File: javascripts/services/ContactService.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    const getFbContacts = (userUid) => {
+        let contacts = [];
+        return $q((resolve, reject) => {
+            $http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+                let fbContacts = results.data;
+                Object.keys(fbContacts).forEach((key) => {
+                    fbContacts[key].id = key;
+                    contacts.push(fbContacts[key]);
+                    resolve(contacts);
```
Comment (rousell) [inline javascripts/services/ContactService.js]: this resolution for your contact should be outside of your looping over of your Object.keys

**Comment by gib-bowden** [comment]
@rousell fixed the bug here and added some validation visuals for the fields. Also opened a pull request for part4, functionality is hopefully done, but I'm going to play around with some styling stuff

**Comment by rousell** [comment]
@gib-bowden really nice job! 👍

**Pull request merged**